### PR TITLE
test(migrations): Check migration state aligns to query schema

### PR DESCRIPTION
**Pull request by lynnagara**
Test that the tables generated by the new migrations are aligned
to the query schema. This checks that the tables generated by the old
migration system and the new migration system are the same.

### CODE REVIEW

File: tests/migrations/test_runner.py [hunk 1]
```diff
@@ -56,3 +57,29 @@ def test_version() -> None:
     assert runner._get_next_version(migration_key) == 2
     runner._update_migration_status(migration_key, Status.COMPLETED)
     assert runner._get_next_version(migration_key) == 3
+
+
+def test_no_schema_differences() -> None:
+    from snuba.datasets.schemas.tables import TableSchema
+    from snuba.datasets.storages import StorageKey
+    from snuba.migrations.parse_schema import get_local_schema
```
Comment (fpacifici) [inline tests/migrations/test_runner.py:65]: Why are these imported in the method ? Not sure I get if there is a circular dependency.

File: tests/migrations/test_runner.py [hunk 2]
```diff
@@ -56,3 +57,29 @@ def test_version() -> None:
...
+    runner = Runner()
+    runner.run_all()
+
+    for storage_key in storages_to_test:
+        storage = get_storage(storage_key)
+        conn = storage.get_cluster().get_query_connection(
+            ClickhouseClientSettings.MIGRATE
+        )
+
+        for schema in storage.get_schemas()._get_unique_schemas():
```
Comment (fpacifici) [inline tests/migrations/test_runner.py:78]: please make get_unique_schemas fully public if you need to access it from outside the DatasetSchemas class and it is not just an implementation detail. `get_unique_schemas`.

File: tests/migrations/test_runner.py [hunk 3]
```diff
@@ -56,3 +57,29 @@ def test_version() -> None:
...
+            if not isinstance(schema, TableSchema):
+                continue
+
+            table_name = schema.get_local_table_name()
+            local_schema = get_local_schema(conn, table_name)
+
+            assert schema.get_column_differences(local_schema) == []
```
Comment (fpacifici) [inline tests/migrations/test_runner.py:85]: We should have a clear error message here that explains why the test is failing and what the difference is. People will add migrations and this is their sanity test. We should make it easy to interpret.

**Review by fpacifici — CHANGES_REQUESTED** [review]
Thanks, this is cool. Just a couple of minor changes.

File: tests/migrations/test_runner.py (hunk 1 above)
Comment (lynnagara) [reply tests/migrations/test_runner.py]: There isn't a circular dependency, I forgot to move it back up

File: tests/migrations/test_runner.py (hunk 2 above)
Comment (lynnagara) [reply tests/migrations/test_runner.py]: Updated

**Review by fpacifici — APPROVED** [review]

**Pull request merged**